### PR TITLE
GTP support for KGS (kgs-chat and kgs time setting)

### PR DESCRIPTION
**Pull request by petgo3**
As i work on a windows maschine (and i'm not so used to git) there is a lot of other stuff.
The change is only in GTP.cpp

### CODE REVIEW

File: cpp/gtp.cpp [hunk 1]
```diff
@@ -1029,67 +1030,126 @@ int MainCmds::gtp(int argc, const char* const* argv) {
...
+		double byoYomiTime;
+		int originalNumPeriods;
+		if (pieces.size() != 4
+			|| pieces[0] != "byoyomi"
+			|| !Global::tryStringToDouble(pieces[1], mainTime)
+			|| !Global::tryStringToDouble(pieces[2], byoYomiTime)
+			|| !Global::tryStringToInt(pieces[3], originalNumPeriods)
+			) {
+			if (pieces.size() != 2
+				|| pieces[0] != "absolut"
```
Comment (lightvector) [inline cpp/gtp.cpp]: Is "absolut" a typo?

**Comment by lightvector** [comment]
Sorry, had not gotten around to take a look until now.

I think I'm not going to accept this pull-request as-is, due to all the random unrelated files that got committed. Also there are some style details that need to be cleaned up, for example like it looks like the part that you edited now differs in indentation style from the rest of the code. And maybe there are a few minor other things to polish up, like briefly documenting the KGS GTP extensions in https://github.com/lightvector/KataGo/blob/master/GTP_Extensions.md, perhaps simply as a link to the associated KGS documentation.

Still, thanks for submitting this! when I do get the chance to work on these details this myself, I'd be more than happy to use your code as a guide if you're okay with that, or at least as a reminder of the semantics of the commands needed to make KGS happy. Or if you really care, you could try to submit a new pull request and take care of these details yourself.

File: cpp/gtp.cpp (hunk 1 above)
Comment (petgo3) [reply cpp/gtp.cpp]: Oops, i confess i did'nt test "absolute". It's a typo, i looked it up in the Doku (kgsGtp.xhtml)

**Comment by petgo3** [comment]
Im quite sorry for all this coding mess, but a lot of it comes from working in windows with visual studio. E.g. formatting. Feel free to decline the pull request and just take some lines of code. If you like i can provide a kgsgtp documentation...Best regardsPeterAm 17.08.2019 16:48 schrieb lightvector <notifications@github.com>:Sorry, had not gotten around to take a look until now.
I think I'm not going to accept this pull-request as-is, due to all the random unrelated files that got committed. Also there are some style details that need to be cleaned up, for example like it looks like the part that you edited now differs in indentation style from the rest of the code. And maybe there are a few minor other things to polish up, like briefly documenting the KGS GTP extensions in https://github.com/lightvector/KataGo/blob/master/GTP_Extensions.md, perhaps simply as a link to the associated KGS documentation.
Still, thanks for submitting this! when I do get the chance to work on these details this myself, I'd be more than happy to use your code as a guide if you're okay with that, or at least as a reminder of the semantics of the commands needed to make KGS happy. Or if you really care, you could try to submit a new pull request and take care of these details yourself.

**Comment by lightvector** [comment]
No problem, this is already very helpful. Where can I find the documentation about kgsgtp?

File: cpp/gtp.cpp (hunk 1 above)
Comment (petgo3) [reply cpp/gtp.cpp]: Dokumentation for kgsgtp: http://www.weddslist.com/kgs/how/kgsGtp.html

Comment (petgo3) [reply cpp/gtp.cpp]: by the way, thx for all the excellent work (hurray!)

**Comment by petgo3** [comment]
I close this request, just to keep project clean.

**Pull request closed without merging**